AppCenter/AppCenter/Internals/Context/Device/MSDeviceTracker.m

```diff
@@ -374,21 +374,31 @@ - (NSNumber *)timeZoneOffset:(NSTimeZone *)timeZone {
 }
 
 - (NSString *)screenSize {
-
 #if TARGET_OS_OSX
-  NSScreen *focusScreen = [NSScreen mainScreen];
-  CGFloat scale = focusScreen.backingScaleFactor;
-  CGSize screenSize = [focusScreen frame].size;
+  NSSize screenSize = [NSScreen mainScreen].frame.size;
+  return [NSString stringWithFormat:@"%dx%d", (int)screenSize.width, (int)screenSize.height];
 #elif TARGET_OS_MACCATALYST
-  // This returns scale value based on scale values iOS.
-  // So we need to use base value for macOS.
-  CGFloat scale = 3;
-  CGSize screenSize = [UIScreen mainScreen].bounds.size;
+
+  // This would report screen resolution as shown in display settings ('Looks like' field in scaling tab).
+  id screen = [NSClassFromString(@"NSScreen") valueForKey:@"mainScreen"];
+  if (screen == nil) {
+    CGSize screenSize = [UIScreen mainScreen].nativeBounds.size;
+    return [NSString stringWithFormat:@"%dx%d", (int)screenSize.width, (int)screenSize.height];
+  }
+  SEL selector = NSSelectorFromString(@"frame");
+  NSInvocation *invocation =
+      [NSInvocation invocationWithMethodSignature:[[screen class] instanceMethodSignatureForSelector:selector]];
+  [invocation setSelector:selector];
+  [invocation setTarget:screen];
+  [invocation invoke];
+  CGRect frame;
+  [invocation getReturnValue:&frame];
+  return [NSString stringWithFormat:@"%dx%d", (int)frame.size.width, (int)frame.size.height];
```

Comment:
For macos and maccatalyst it shows "screenSize" : "1440x900" (actual is "2880x1800"), while for ios it is "1334x750" (actual is "1334x750").
If this approach for macos &maccatalyst is showing correct numbers why we can't multiply them by the scale factor as we did before? Does it have any caveats?
I think we should handle it following the same logic for all the platforms.